### ISSUE TITLE
Ignore newlines inside paragraphs

### ISSUE THREAD

**Issue by davizin-86** [flutter/flutter_markdown#44]
Only a blank line should mark the end of a paragraph.
It is awfull to see arbitrary line breaks inside a paragraph.

Thanks for your work.

-- david

It seems to be enough with:

builder.dart

visitText(...) {
...
text: text.text.trim().replaceAll("\n", " "),

**Comment by maral** [comment]
Little upgrade to prevent deleting double new-lines as well:
```
string.trim().replaceAllMapped(RegExp("([^\n])\n([^\n])"), (match) {
    return "${match.group(1)} ${match.group(2)}";
});
```

**Comment by PrimaryFeather** [comment]
I was equally surprised by this behavior! Normally, only "two spaces + newline" should trigger a line break inside a paragraph. Interestingly, exactly this behavior is even checked by a unit test: https://github.com/flutter/flutter_markdown/blob/898ed051362a5cf50d9712e35ac020db9585bd0b/test/flutter_markdown_test.dart#L65-L71

However, the "negative" test (no line break if it's just a single "newline") is lacking.
It would be great if this could be fixed in a future release, or at least discussed here.

Thanks a lot in advance, and thanks everyone involved in this excellent package!

**Comment by domesticmouse** [comment]
`flutter_markdown`'s markdown parser is `markdown`. Can you please re-raise this issue [over here](https://github.com/dart-lang/markdown/issues) instead? Thanks!

**Comment by PrimaryFeather** [comment]
Ah, of course — makes sense! I'll do that. Cheers!

**Comment by PrimaryFeather** [comment]
Hm, actually this *does* seem to be an issue of _flutter_markdown_, after all.

I just created the following quick command-line app in dart:

```dart
import 'package:markdown/markdown.dart';

void main() {
  final markdown = "Line 1.\nLine 2.\n\nLine 3.\nLine 4.";
  print(markdownToHtml(markdown));
}
```

The result is this:

```html
<p>Line 1.
Line 2.</p>
<p>Line 3.
Line 4.</p>
```

In other words, the pure _markdown_ library creates two paragraphs, as expected, without any `<br>` elements.

Could you maybe take another look? Or is this a misunderstanding on my side?
Thanks a lot in advance!

**Comment by PrimaryFeather** [comment]
Any comments on this? Should I open up a new issue?

**Comment by lzhuor** [comment]
Is this issue #221 also relevant to the problem?

**Comment by lzhuor** [comment]
Hi @PrimaryFeather @davizin-86 , I found that it's indeed a problem from `flutter_markdown`. The root cause is how the current line splitting logic is written.

flutter_markdown/lib/src/widget.dart, L:163
```dart
final List<String> lines = widget.data.split(RegExp(r'\r?\n'));
```

**Comment by PrimaryFeather** [comment]
Just for reference: this problem was fixed in the above pull request (#218) which is part of version 0.4.0.

**Comment by domesticmouse** [comment]
Thanks for the investigation @PrimaryFeather, much appreciated
